### PR TITLE
Go: Rename `UntrustedFlowSource` to `RemoteFlowSource` to match other language libraries

### PR DESCRIPTION
**Pull request by owen-mc**
Go was the the only language library which wasn't using the name `RemoteFlowSource`, so this seemed worth fixing for the sake of consistency.

Related renamings have also been done. Some classes which should have been private have been made private while we are deprecating the old name anyway. Appropriate instructions have been put in the deprecation comments.

### CODE REVIEW

File: go/ql/lib/semmle/go/security/CommandInjectionCustomizations.qll [hunk 1]
```diff
@@ -29,8 +29,13 @@ module CommandInjection {
    */
   abstract class Sanitizer extends DataFlow::Node { }
 
+  /**
+   * DEPRECATED: Use `RemoteFlowSource` or `Source` instead.
+   */
+  deprecated class UntrustedFlowAsSource = RemoteFlowAsSource;
```
Comment (mbg) [inline go/ql/lib/semmle/go/security/CommandInjectionCustomizations.qll:35]: Does this comment make sense, given that `RemoteFlowAsSource` is marked as `private` here?

File: go/ql/lib/semmle/go/security/CommandInjectionCustomizations.qll [hunk 2]
```diff
@@ -29,8 +29,13 @@ module CommandInjection {
...
+
   /** A source of untrusted data, considered as a taint source for command injection. */
-  class UntrustedFlowAsSource extends Source instanceof UntrustedFlowSource { }
+  private class RemoteFlowAsSource extends Source instanceof RemoteFlowSource { }
```
Comment (mbg) [inline go/ql/lib/semmle/go/security/CommandInjectionCustomizations.qll:38]: Why `private`?

File: go/ql/lib/semmle/go/security/LogInjectionCustomizations.qll [hunk 3]
```diff
@@ -25,8 +25,13 @@ module LogInjection {
    */
   abstract class Sanitizer extends DataFlow::Node { }
 
+  /**
+   * DEPRECATED: Use `RemoteFlowSource` or `Source` instead.
+   */
+  deprecated class UntrustedFlowAsSource = RemoteFlowAsSource;
```
Comment (mbg) [inline go/ql/lib/semmle/go/security/LogInjectionCustomizations.qll:31]: Does the comment make sense, given that the replacement is `private` now?

File: go/ql/lib/semmle/go/security/OpenUrlRedirectCustomizations.qll [hunk 4]
```diff
@@ -42,11 +42,16 @@ module OpenUrlRedirect {
     abstract predicate hasTaintStep(DataFlow::Node pred, DataFlow::Node succ);
   }
 
+  /**
+   * DEPRECATED: Use `RemoteFlowSource` or `Source` instead.
+   */
+  deprecated class UntrustedFlowAsSource = RemoteFlowAsSource;
```
Comment (mbg) [inline go/ql/lib/semmle/go/security/OpenUrlRedirectCustomizations.qll:48]: Does this make sense given that the replacement is private?

File: go/ql/lib/semmle/go/security/ReflectedXssCustomizations.qll [hunk 5]
```diff
@@ -34,10 +34,15 @@ module ReflectedXss {
     }
   }
 
+  /**
+   * DEPRECATED: Use `RemoteFlowSource` or `Source` instead.
+   */
```
Comment (mbg) [inline go/ql/lib/semmle/go/security/ReflectedXssCustomizations.qll:39]: Does this make sense given that the replacement is private?

File: go/ql/lib/semmle/go/security/RequestForgeryCustomizations.qll [hunk 6]
```diff
@@ -32,10 +32,15 @@ module RequestForgery {
   /** An outgoing sanitizer edge for request forgery vulnerabilities. */
   abstract class SanitizerEdge extends DataFlow::Node { }
 
+  /**
+   * DEPRECATED: Use `RemoteFlowSource` or `Source` instead.
+   */
```
Comment (mbg) [inline go/ql/lib/semmle/go/security/RequestForgeryCustomizations.qll:37]: Does this make sense given that the replacement is private?

File: go/ql/lib/semmle/go/security/SqlInjectionCustomizations.qll [hunk 7]
```diff
@@ -25,8 +25,13 @@ module SqlInjection {
    */
   abstract class Sanitizer extends DataFlow::Node { }
 
+  /**
+   * DEPRECATED: Use `RemoteFlowSource` or `Source` instead.
+   */
```
Comment (mbg) [inline go/ql/lib/semmle/go/security/SqlInjectionCustomizations.qll:30]: Does this make sense given that the replacement is private?

File: go/ql/lib/semmle/go/security/TaintedPathCustomizations.qll [hunk 8]
```diff
@@ -44,8 +44,13 @@ module TaintedPath {
     }
   }
 
+  /**
+   * DEPRECATED: Use `RemoteFlowSource` or `Source` instead.
+   */
+  deprecated class UntrustedFlowAsSource = RemoteFlowAsSource;
```
Comment (mbg) [inline go/ql/lib/semmle/go/security/TaintedPathCustomizations.qll:50]: Does this make sense given that the replacement is private?

File: go/ql/lib/semmle/go/security/XPathInjectionCustomizations.qll [hunk 9]
```diff
@@ -24,8 +24,13 @@ module XPathInjection {
    */
   abstract class Sanitizer extends DataFlow::ExprNode { }
 
+  /**
+   * DEPRECATED: Use `RemoteFlowSource` or `Source` instead.
+   */
```
Comment (mbg) [inline go/ql/lib/semmle/go/security/XPathInjectionCustomizations.qll:29]: Does this make sense given that the replacement is private?

File: go/ql/src/Security/CWE-640/EmailInjectionCustomizations.qll [hunk 10]
```diff
@@ -16,8 +16,13 @@ module EmailInjection {
    */
   abstract class Sink extends DataFlow::Node { }
 
+  /**
+   * DEPRECATED: Use `RemoteFlowSource` or `Source` instead.
+   */
```
Comment (mbg) [inline go/ql/src/Security/CWE-640/EmailInjectionCustomizations.qll:21]: Does this make sense given that the replacement is private?

File: go/ql/src/experimental/CWE-918/SSRF.qll [hunk 11]
```diff
@@ -87,10 +87,15 @@ module ServerSideRequestForgery {
   /** An outgoing sanitizer edge for request forgery vulnerabilities. */
   abstract class SanitizerEdge extends DataFlow::Node { }
 
+  /**
+   * DEPRECATED: Use `RemoteFlowSource` or `Source` instead.
+   */
```
Comment (mbg) [inline go/ql/src/experimental/CWE-918/SSRF.qll:92]: Does this make sense given that the replacement is private?

**Review by mbg — COMMENTED** [review]
Thanks for doing this to bring Go in line with other languages here! This generally all looks good. I just have some minor comments and a recurring question about a bunch of the comments on the now `deprecated` classes, which suggests using a now `private` replacement. Feel free to answer one and ignore the others if that's fine as is.

> Some classes which should have been private have been made private while we are deprecating the old name anyway.

Why should they be `private`?

File: go/ql/lib/semmle/go/security/CommandInjectionCustomizations.qll (hunk 1 above)
Comment (owen-mc) [reply go/ql/lib/semmle/go/security/CommandInjectionCustomizations.qll:35]: The comment doesn't mention `RemoteFlowAsSource`, it mentions `RemoteFlowSource`.

File: go/ql/lib/semmle/go/security/CommandInjectionCustomizations.qll (hunk 2 above)
Comment (owen-mc) [reply go/ql/lib/semmle/go/security/CommandInjectionCustomizations.qll:38]: It just doesn't make much sense to make this public. It isn't really of any use for external code to reference it, and will probably just lead to confusion if it is done by mistake.

File: go/ql/lib/semmle/go/security/CommandInjectionCustomizations.qll (hunk 1 above)
Comment (mbg) [reply go/ql/lib/semmle/go/security/CommandInjectionCustomizations.qll:35]: I genuinely did not notice that when reading any of the comments I flagged up! My bad then, the comments are all fine then.

**Pull request merged**